src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -380,3 +382,28 @@ export const selectPortfolioIsEmpty = createSelector(
   selectPortfolioAssetIds,
   (assetIds): boolean => !assetIds.length
 )
+
+export const selectPortfolioAccounts = (state: ReduxState) => state.portfolio.accounts.byId
+
+export const selectPortfolioAssetAccounts = createSelector(
+  selectPortfolioAccounts,
+  (_state: ReduxState, assetId: CAIP19) => assetId,
+  (portfolioAccounts, assetId): AccountSpecifier[] => {
```

Comment:
Non-blocking improvement proposal for selector perf/readability.
Realistically, performance won't be an issue since you wouldn't have many accounts with a single asset (probably never even two digits), so the proposal is mostly about readability:

```
(portfolioAccounts, assetId) =>
    Object.keys(portfolioAccounts).filter(accountSpecifier =>
      portfolioAccounts[accountSpecifier].find(accountAssetId => accountAssetId === assetId)
    )
```

And with this construct, no need for writing explicit typings 🎉  what do you think?

Reply:
actually yeah this is much nicer, thank you, i'll make this change 🤘